### PR TITLE
Increase required Go version to 1.19 in tutorial

### LINKED ISSUE #167: Update Go version to 1.18 or 1.19

### PR DESCRIPTION
**Pull request by marten-seemann**
resolves #167

### CODE REVIEW

**Comment by marten-seemann** [comment]
This is already outdated. We should find a way to write this such that we don’t have to update the version every 6 months.

**Comment by BigLep** [comment]
2022-08-19 conversation: will cover in the one-off docs triage happening week of 2022-08-22.

**Comment by MarcoPolo** [comment]
Maybe we can just say the latest version - 1?

**Comment by MarcoPolo** [comment]
also we could figure this out at runtime with https://go.dev/dl/?mode=json and https://go.dev/VERSION?m=text

File: content/tutorials/getting-started/go.md [hunk 1]
```diff
@@ -15,26 +15,26 @@ messages between them.
 ## Install Go
 
 go-libp2p recommends the use of a Go version that includes the [modules feature](https://github.com/golang/go/wiki/Modules),
-which means you need to have a version of Go that is at least 1.11.
+which means you need to have a version of Go that is at least 1.16.
```
Comment (DannyS03) [inline content/tutorials/getting-started/go.md:18]:
```suggestion
which means you need to have a version of Go that is at least 1.19.
```

File: content/tutorials/getting-started/go.md [hunk 2]
```diff
@@ -15,26 +15,26 @@ messages between them.
...
 
 You can install a recent version of Go by following the [official installation instructions](https://golang.org/doc/install).
 
-Once installed, you should be able to run `go version` and see a version >= 1.11, for example:
+Once installed, you should be able to run `go version` and see a version >= 1.16, for example:
```
Comment (DannyS03) [inline content/tutorials/getting-started/go.md:22]:
```suggestion
Once installed, you should be able to run `go version` and see a version >= 1.19, for example:
```

File: content/tutorials/getting-started/go.md [hunk 3]
```diff
@@ -15,26 +15,26 @@ messages between them.
...
 
 ```sh
 $ go version
-go version go1.12 darwin/amd64
+go version go1.17.5 darwin/amd64
```
Comment (DannyS03) [inline content/tutorials/getting-started/go.md:26]:
```suggestion
go version go1.19 darwin/arm64
```

File: content/tutorials/getting-started/go.md [hunk 4]
```diff
@@ -51,7 +51,7 @@ initialized and the version of Go you're using, for example:
 $ cat go.mod
 module github.com/user/go-libp2p-tutorial
 
-go 1.12
+go 1.17
```
Comment (DannyS03) [inline content/tutorials/getting-started/go.md:54]:
```suggestion
go 1.19
```

**Review by DannyS03 — CHANGES_REQUESTED** [review]
Based on https://github.com/libp2p/docs/issues/167, suggesting the following updates directly on this open PR.

To also bounce off the discussion on this PR and during the docs triage, we can add automation where the value is pulled in and injected onto the page, or, at the very least, an alert for new releases

**Comment by DannyS03** [comment]
@p-shahi @marten-seemann had added suggestions to this open PR to update the version

File: content/tutorials/getting-started/go.md [hunk 5]
```diff
@@ -15,26 +15,26 @@ messages between them.
 ## Install Go
 
 go-libp2p recommends the use of a Go version that includes the [modules feature](https://github.com/golang/go/wiki/Modules),
```
Comment (marten-seemann) [inline content/tutorials/getting-started/go.md:17]: Every Go version you'd realistically use nowadays has modules support. We should remove this sentence.

Comment (DannyS03) [inline content/tutorials/getting-started/go.md:17]:
```suggestion
```

Comment (DannyS03) [reply content/tutorials/getting-started/go.md:17]: added suggestion

**Review by marten-seemann — COMMENTED** [review]
LGTM. Can't approve, since I'm the original author of this PR.

**Review by DannyS03 — APPROVED** [review]

**Pull request merged**